File: Love Calculator/Love_Calc_Algorithm.py
'''
This is a love calculator algorithm based on Name of two person. This is only and only for fun purpsose.

'''


# Enter the name
a = input("Enter your First name: ")
b = input("Enter your partner's First name: ")

if not a.isalpha() or not b.isalpha():
    print("\n\nINVALID INPUT")

a = a.lower()
b = b.lower()

love = {}

for i in a:
    love[i] = love.get(i,0)+1

for i in b:
    love[i] = love.get(i,0)+1

percent = 0

for s,i in love.items():
    if i>1:
        percent += (ord(s)-ord('a'))*i

if percent>=100:
    print(f"\n\n Wow, You two are Soulmates. With perfect 100 % love.")
elif 80<= percent < 100:
    print(f"\n\n That's some serious love right there with {percent} %.")

elif 60<= percent <80:
    print(f"\n\n You have good compatability, with love score of {percent} %.")

else:
    print(f"\n\n You will face challenges but Hey, your story will be an amazing one my friend. Your love score is {percent} %.")

File: Love Calculator/Love_Calc_Algorithm.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class LoveCalculator {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter your First name: ");
        String a = scanner.nextLine().toLowerCase();
        System.out.print("Enter your partner's First name: ");
        String b = scanner.nextLine().toLowerCase();

        if (!a.matches("[a-zA-Z]+") || !b.matches("[a-zA-Z]+")) {
            System.out.println("\nINVALID INPUT");
            return;
        }

        Map<Character, Integer> love = new HashMap<>();

        for (char c : a.toCharArray()) {
            love.put(c, love.getOrDefault(c, 0) + 1);
        }

        for (char c : b.toCharArray()) {
            love.put(c, love.getOrDefault(c, 0) + 1);
        }

        int percent = 0;

        for (Map.Entry<Character, Integer> entry : love.entrySet()) {
            char s = entry.getKey();
            int i = entry.getValue();
            if (i > 1) {
                percent += (s - 'a') * i;
            }
        }

        if (percent >= 100) {
            System.out.println("\nWow, You two are Soulmates. With a perfect 100 % love.");
        } else if (percent >= 80) {
            System.out.println("\nThat's some serious love right there with " + percent + " %.");
        } else if (percent >= 60) {
            System.out.println("\nYou have good compatibility, with a love score of " + percent + " %.");
        } else {
            System.out.println("\nYou will face challenges, but your story will be an amazing one, my friend. Your love score is " + percent + " %.");
        }
    }
}

